ARCHITECTURE.md

```diff
@@ -50,18 +52,26 @@ auxiliary ones. Packages are a unit of dependency in Cargo, and when
 package `foo` depends on package `bar`, that means that each target
 from `foo` needs the library target from `bar`.
 
-`PackageId` is the unique identifier of a (possibly remote)
+[`PackageId`] is the unique identifier of a (possibly remote)
 package. It consist of three components: name, version and source
 id. Source is the place where the source code for package comes
 from. Typical sources are crates.io, a git repository or a folder on
 the local hard drive.
 
-`Resolve` is the representation of a directed acyclic graph of package
-dependencies, which uses `PackageId`s for nodes. This is the data
+[`Resolve`] is the representation of a directed acyclic graph of package
+dependencies, which uses [`PackageId`]s for nodes. This is the data
 structure that is saved to the lock file. If there is no lock file,
 Cargo constructs a resolve by finding a graph of packages which
 matches declared dependency specification according to semver.
 
+[`Config`]: https://docs.rs/cargo/0.33.0/cargo/util/config/struct.Config.html
+[`shell`]: https://docs.rs/cargo/0.33.0/cargo/util/config/struct.Config.html#method.shell
+[`Workspace`]: https://docs.rs/cargo/0.33.0/cargo/core/struct.Workspace.html
+[`Package`]: https://docs.rs/cargo/0.33.0/cargo/core/package/struct.Package.html
+[`Target`]: https://docs.rs/cargo/0.33.0/cargo/core/manifest/struct.Target.html
+[`PackageId`]: https://docs.rs/cargo/0.33.0/cargo/core/package_id/struct.PackageId.html
+[`Resolve`]: https://docs.rs/cargo/0.33.0/cargo/core/struct.Resolve.html
```

Comment:
I was thinking it would be best if these too were relative to, say, `src/cargo/core/resolver/resolve.rs`.

Reply:
I think rendered document at docs.rs would be more readable, and it would be more useful thanks to the cross-links between structs, methods, etc.

Reply:
Yeah, I know what you mean.  Maybe then, to avoid having a stale version or the maintenance of updating the version, we should use https://docs.rs/cargo/latest/cargo/core/struct.Resolve.html?

Reply:
> Maybe then, to avoid having a stale version or the maintenance of updating the version, we should use https://docs.rs/cargo/latest/cargo/core/struct.Resolve.html?

I made a silly mistake. Fixed :smile: